nms/app/packages/magmalte/app/views/equipment/GatewayDetailConfigEdit.js

```diff
@@ -580,14 +619,24 @@ export function EPCEdit(props: Props) {
   const ctx = useContext(GatewayContext);
 
   const handleEPCChange = (key: string, val) => {
-    setEPCConfig({...EPCConfig, [key]: val});
+    if (key === 'nat_enabled' && val) {
+      setEPCConfig({
+        ...EPCConfig,
+        [key]: val,
+        ['sgi_management_iface_vlan']: '',
+      });
+    } else {
+      setEPCConfig({...EPCConfig, [key]: val});
+    }
   };
 
   const [EPCConfig, setEPCConfig] = useState<gateway_epc_configs>(
     props.gateway?.cellular.epc || DEFAULT_GATEWAY_CONFIG.cellular.epc,
   );
-
   const onSave = async () => {
+    if (EPCConfig.nat_enabled) {
+      handleEPCChange('sgi_management_iface_vlan', '');
```

Comment:
Might be cleaner to directly set the epcConfig here rather than call handleEpcChange. Kind of confusing to see a handleEpcChange being called during Save.